dm/worker/source_worker.go

```diff
@@ -186,11 +218,54 @@ func (w *Worker) Close() {
 	}
 
 	w.closed.Store(true)
+
+	w.sourceDB.Close()
```

Comment:
`w.closed.Store(true)` is to make sure happened-before relationship (though it failed to protect the concurrent accessing to `w.sourceDB` for now)

Reply:
above comment is wrong because I missed `SourceWorker.RWMutex`

https://github.com/pingcap/dm/pull/2076/commits/b3670c92d99450a20ee37e826324f9dd6d086ff0

Reply:
What if `w.sourceDB` is nil?

Reply:
`(*BaseDB).Close()` will check that